tools/windows/install-help/cal/customactiondata.cpp

```diff
@@ -102,74 +100,165 @@ void CustomActionData::Sid(sid_ptr &sid)
+    if (bKey)
     {
-        // explicitly disabled
-        WcaLog(LOGMSG_STANDARD, "SYSPROBE_PRESENT explicitly disabled %S", sysprobePresent.c_str());
-        return true;
+        if ( closedSource == 1 )
+        {
+            WcaLog(LOGMSG_STANDARD, "Closed source already marked accepted");
+        }
+        if ( closedSource == 0 )
+        {
+            WcaLog(LOGMSG_STANDARD, "Closed source already marked disabled");
+        }
+        // else what do we do here?
     }
-    this->_doInstallSysprobe = true;
 
-    if (!this->_propertyView->value(L"NPM", npm))
+   
+    // check the ADDLOCAL flag
+    if(!setEnabledFlag)
     {
-        WcaLog(LOGMSG_STANDARD, "NPM property not present");
+        if(this->_propertyView->value(L"ADDLOCAL", addlocal)) 
+        {
+            // Convert to upper when normalizing string for comparison
+            // https://learn.microsoft.com/en-us/previous-versions/visualstudio/visual-studio-2015/code-quality/ca1308-normalize-strings-to-uppercase
+            std::transform(addlocal.begin(), addlocal.end(), addlocal.begin(), toupper);
+            if (addlocal.find(L"ALL") != std::wstring::npos ||
+                addlocal.find(L"NPM") != std::wstring::npos)
+            {
+                WcaLog(LOGMSG_STANDARD, "Found addlocal key %S.  Allowing closed source", addlocal.c_str());
+                WcaLog(LOGMSG_STANDARD, "Installation is no longer controlled via Windows Features.  Please update install tools");
+                newEnabledFlag = true;
+                setEnabledFlag = true;
+            }
+            else
+            {
+                WcaLog(LOGMSG_STANDARD, "ADDLOCAL key does not contain all/NPM (%S)", addlocal.c_str());
+            }
+        }
     }
-    else
+    if (!setEnabledFlag)
     {
-        WcaLog(LOGMSG_STANDARD, "NPM enabled via NPM property");
-        this->_ddnpmPresent = true;
+        std::wstring npmProperty;
+        if (this->_propertyView->value(L"NPM", npmProperty))
+        {
+            // if this property is set to anything besides the empty string,
+            // the previous installers would install NPM.  That's good enough for us
+
+            WcaLog(LOGMSG_STANDARD, "NPM key is present and (%S)", npmProperty.c_str());
+            if (npmProperty.length() > 0)
+            {
+                WcaLog(LOGMSG_STANDARD, "Allowing closed source because NPM flag is set");
+                newEnabledFlag = true;
+                setEnabledFlag = true;
+            }
+        }
     }
 
-    if (this->_propertyView->value(L"NPMFEATURE", npmFeature))
+    if (!setEnabledFlag)
     {
-        // this property is set to "on" or "off" depending on the desired installed state
-        // of the NPM feature.
-        WcaLog(LOGMSG_STANDARD, "NPMFEATURE key is present and (%S)", npmFeature.c_str());
-        if (_wcsicmp(npmFeature.c_str(), L"on") == 0)
+        
+        if (this->_propertyView->value(L"CLOSEDSOURCE", csProperty))
```

Comment:
should this be `ALLOWCLOSEDSOURCE` ?

Reply:
No, it should be that.  It's provided in the custom action string as `CLOSEDSOURCE`. 
https://github.com/DataDog/datadog-agent/pull/15263/files#diff-d6158508805a99eaed5c28596a83246becbca20978d481d7d87d14cc754e128aR710-R713

The reason the name is different is that there's different properties flying around in the xml, that get changed at different times during different conditionals; I wanted something non-ambiguous.  Although, it is clearly not straightforward.

Reply:
OK.  so the above link, while it points to the right place, won't take you there.

But, the property is supplied here:
```
            CLOSEDSOURCE=[ALLOWCLOSEDSOURCE]
            DDNPM_INSTALLED=[DD_FILTER_INSTALLED]
```